bulk_indexer_test.go

```diff
@@ -0,0 +1,100 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package docappender_test
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/elastic/go-docappender/v2"
+	"github.com/elastic/go-docappender/v2/docappendertest"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBulkIndexer(t *testing.T) {
```

Comment:
[**For reviewers**] I wanted to write a black box test for the bulk indexer with large data (the current bug would only be detected if the document count was high). This led me to write this test which is slightly different than the appended test. Note that we don't currently test when compression is enabled -- this should also be done and I will create a followup for that later.

Reply:
Appreciate the additional test here!

Reply:
@simitt Are you referring to adding more tests (with compression) in this PR itself?

Reply:
Added tests with different compression levels.

Reply:
I meant to say that I appreciate that you have added the _TestBulkIndexer_ test.